src/helpers/shared/pagination.ts

```diff
@@ -0,0 +1,85 @@
+import { NavigateFunction, Location } from 'react-router-dom';
+
+export interface PaginationDefaultI {
+  limit?: number;
+  offset?: number;
+  count?: number; // from API
+  itemCount?: number; // for PF pagination
+}
+
+export const defaultSettings: Required<PaginationDefaultI> = {
+  limit: 20,
+  offset: 0,
+  itemCount: 0,
+  count: 0,
+};
+
+export const defaultAdminSettings = {
+  ...defaultSettings,
+  limit: 50,
+};
+
+export const defaultCompactSettings = {
+  ...defaultSettings,
+  limit: 10,
+};
+
+export const calculatePage = (limit = defaultSettings.limit, offset = 0) => Math.floor(offset / limit) + 1;
+
+export const calculateOffset = (page = 1, limit = defaultSettings.limit) => (page - 1) * limit;
+
+export const syncDefaultPaginationWithUrl = (location: Location, navigate: NavigateFunction, defaultPagination: PaginationDefaultI) => {
+  if (!defaultPagination) {
+    defaultPagination = defaultSettings;
+  }
+  const searchParams = new URLSearchParams(location.search);
+
+  let limit = parseInt(searchParams.get('per_page') as string);
+  let page = parseInt(searchParams.get('page') as string);
+
+  if (isNaN(limit) || limit <= 0) {
+    limit = defaultPagination.limit as number;
```

Comment:
I don't think we need the `as number` now that the default object has all attributes required.